src/normalization.rs

```diff
@@ -0,0 +1,163 @@
+//! Normalizer trait and default implementations.
+//!
+//! Normalization is how strings are normalized into characters you want to
+//! consider equal. For example, if you want a case insensitive comparison,
+//! you might consider the [LowerCaseNormalizer].
+//!
+//! ```
+//! # use fuzzywuzzy::normalization::{Normalizer, LowerCaseNormalizer, FormCNormalization, MultipleNormalizer};
+//! assert_eq!(LowerCaseNormalizer.normalize("this STRING"), LowerCaseNormalizer.normalize("THIS string"));
+//! let a1 = "ä"; // U+00E4
+//! let a2 = "ä"; // U+0061 + U+0308
+//! let a3 = "Ä"; // U+0041 + U+0308
+//! assert_ne!(a1, a2);
+//! assert_eq!(FormCNormalization.normalize(a2), a1);
+//! let multiple_normalizers = MultipleNormalizer::with(vec![Box::new(LowerCaseNormalizer), Box::new(FormCNormalization)]);
+//! assert_eq!(multiple_normalizers.normalize(a3), a1);
+//! ```
+
+/// Represents a strategy for normalizing string characters into a canonical value of their equivalence class.
+///
+/// i.e., in a case-insensitive context, 'a' might be the canonical value for the equivalence class of ASCII A's: `['a', 'A']`.
+pub trait Normalizer {
+    fn normalize(&self, s: &str) -> String;
+}
+
+impl<F: Fn(&str) -> String> Normalizer for F {
+    fn normalize(&self, s: &str) -> String {
+        self(s)
+    }
+}
+
+/// Doesn't modify any characters. The Identity-transform [Normalizer].
+pub struct PassthroughNormalizer;
+
+impl Normalizer for PassthroughNormalizer {
+    fn normalize(&self, s: &str) -> String {
+        s.into()
+    }
+}
+
+// ew, need a better name
```

Comment:
Off the top of my head:
* `ChainedNormalizer`
* `Normalizers`
* `ComposedNormalizer`
* `NormalizerSequence`

Let me know your thoughts. Naming certainly is the hardest task in programming 🙂

Reply:
I like `ComposedNormalizer` or maybe `ComposingNormalizer`?

I had thought of `SequenceNormalizer` but that feels unintuitive (like it might normalize sequences (of what?)) and discarded `NormalizerSequence` for the nebulous reason that I'd like to keep the `*Normalizer` suffix.

Reply:
`ComposedNormalizer` works for me, and it fits within the nebulous pattern 😉